### DISCUSSION TITLE
Custom styled property coercion and templated control

### DISCUSSION THREAD

**Discussion by XDgl233** [AvaloniaUI/Avalonia#13945] [Q&A] [answered]
The problem is mostly described in the title, I have a simple templated control that has 4 textboxes to mimic the Window style of IP address input. I create four styled properties and bind them in two-way mode with those child textboxes.
The problem comes with data validation and coercion. The coercion works correctly with those custom styled properties, but those textboxes's text properties are not updated correctly. I set the coercion rules with the value set to an integer range between 0-255. When I type in the text, like 500, the text will be replaced by 255, but if I continue to type in and the input is now something like 2550, the coercion happens to the custom property and correctly be coerced to 255, but the textbox won't update the new value and display to what user input is.
If I might take a guess of the reason, that when the custom property is already coerced to 255, and new input still will coerce to the same value, thus not fire the event to inform the textbox to update the text value? If so, I still don't have a clue how to manually fire the update event something like OnPropertyChanged() in community toolkit in a templated control.
Any help is greatly appreciated, thanks :)

**Comment by timunie** [comment, marked as answer]
you can override OnProeprtyChanged in your control as well for example
